### PR TITLE
Install guide improvements

### PR DESCRIPTION
**Pull request by shrielenee**
## Purpose of this pull request

This pull request (PR) revises the installation guide to create persona-based flows for contributor, non-contributor, and advanced users.

_Preview build: 137/guides/v2.4/install-gde/install-flow-diagram.html_

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/install-gde/install/cli/dev_options.html
- https://devdocs.magento.com/guides/v2.4/install-gde/prereq/dev_install.html#prerequisites

## Links to Magento source code

-  ...

### CODE REVIEW

File: src/_data/toc/installation-guide.yml [hunk 1]
```diff
@@ -83,10 +76,16 @@ pages:
         - label: Install the Magento software
           url: /install-gde/install/cli/install-cli-install.html
 
-        - label: Contributing developers—update, reinstall Magento
+        - label: (Contributor) Update or reinstall Magento
```
Comment (jeff-matthews) [inline src/_data/toc/installation-guide.yml:79]: This section of the TOC has a lot of tasks now. Maybe we should consider changing the title of this topic to something more generic, like "Contributor workflows"? That might not be the best title, but it's all I can come up with at the moment.

**Review by jeff-matthews — COMMENTED** [review]
I like the consolidation, but I wonder if we should retitle the "Contributor" topic.

File: src/guides/v2.3/install-gde/install/cli/dev_options.md [hunk 2]
```diff
@@ -9,12 +9,14 @@ functional_areas:
...
 *  To [change versions]({{ page.baseurl }}/install-gde/install/cli/dev_downgrade.html) from `develop` to a release version like `2.0.4`, you must uninstall the Magento software and install the released version.
-*  To [add, remove, or update components]({{ page.baseurl }}/install-gde/install/cli/dev_add-update.html), modify `composer.json` and run `composer update` and update the Magento database
-*  To [reinstall the Magento software]({{ page.baseurl }}/install-gde/install/cli/dev_reinstall.html), modify the product version in `composer.json`, run `composer update`, then reinstall the Magento software
+*  To [reinstall the Magento software]({{ page.baseurl }}/install-gde/install/cli/dev_reinstall.html), modify the product version in `composer.json`, run `composer update`, then reinstall the Magento software.
 
  {:.bs-callout-info}
 If you are not a contributing developer, you perform upgrades and upgrades as discussed in [Updating the Magento application and components]({{ page.baseurl }}/comp-mgr/bk-compman-upgrade-guide.html).
```
Comment (keharper) [inline src/guides/v2.3/install-gde/install/cli/dev_options.md:22]: Perhaps one of the instances of "upgrades" should be "updates"?

File: src/guides/v2.3/install-gde/install/cli/dev_options.md [hunk 3]
```diff
@@ -9,12 +9,14 @@ functional_areas:
...
+*  To ensure your system can run the Magento application successfully, [resolve any dependencies]({{ page.baseurl }}/install-gde/install/prepare-install.html) before you proceed with installation.
+*  To [update the Magento software]({{ page.baseurl }}/install-gde/install/cli/dev_update-magento.html), use `git pull origin` and `composer update`, then update the Magento database.
+*  To [add, remove, or update components]({{ page.baseurl }}/install-gde/install/cli/dev_add-update.html), modify `composer.json` and run `composer update` and update the Magento database.
 *  To [change versions]({{ page.baseurl }}/install-gde/install/cli/dev_downgrade.html) from `develop` to a release version like `2.0.4`, you must uninstall the Magento software and install the released version.
```
Comment (keharper) [inline src/guides/v2.3/install-gde/install/cli/dev_options.md:18]: Change `2.0.4` to 2.4.1

File: src/guides/v2.3/install-gde/install/cli/dev_options.md [hunk 4]
```diff
@@ -9,12 +9,14 @@ functional_areas:
...
 
-*  To [update the Magento software]({{ page.baseurl }}/install-gde/install/cli/dev_update-magento.html), use `git pull origin` and `composer update`, then update the Magento database
+*  To get started with your Magento installation or update, ensure your server meets [our system requirements]({{ page.baseurl }}/install-gde/prereq/dev_install.html#prerequisites), [create the Magento file system owner]({{ page.baseurl }}/install-gde/prereq/dev_install.html#prerequisites), [install Composer]({{ page.baseurl }}/install-gde/prereq/dev_install.html#instgde-prereq-compose-install), and [clone the repository]({{ page.baseurl }}/install-gde/prereq/dev_install.html#instgde-prereq-compose-clone).
+*  To ensure your system can run the Magento application successfully, [resolve any dependencies]({{ page.baseurl }}/install-gde/install/prepare-install.html) before you proceed with installation.
+*  To [update the Magento software]({{ page.baseurl }}/install-gde/install/cli/dev_update-magento.html), use `git pull origin` and `composer update`, then update the Magento database.
```
Comment (keharper) [inline src/guides/v2.3/install-gde/install/cli/dev_options.md:16]: Too much detail for a summary topic. Let the linked topic discuss specific commands. Rewrite to something like

To update the Magento software, pull in changes from the Magento 2 GitHub repository, update dependencies, and update the Magento database.

Same comment below.

File: src/_data/toc/installation-guide.yml (hunk 1 above)
Comment (keharper) [reply src/_data/toc/installation-guide.yml:79]: I know you can only do so much at one time, but consider moving update/reinstall info to the Software Update Guide.

**Comment by shrielenee** [comment]
Ok... I did quite a few things. _Maybe_ I went a little too hogwild. You tell me.

- I created two persona-specific flows: Quick start install (non-contributor) and Command Line install (contributor).
- Removed Install options topic because it was a replication of info in Get the Magento software topic (and is linked where needed).
- Did some renaming and moving of topics to fit within this new persona-specific flow approach.

File: src/_data/toc/installation-guide.yml [hunk 5]
```diff
@@ -60,44 +60,36 @@ pages:
...
-
-    - label: Quick start (recommended installation)
+    - label: Quick start install (non-contributor)
       url: /install-gde/composer.html
 
-    - label: Install options
-      url: /install-gde/bk-install-guide.html
-
-    - label: Command Line Installation
+    - label: Command Line install (contributor)
```
Comment (jeff-matthews) [inline src/_data/toc/installation-guide.yml:66]: This change conflates the contributor install method (git-based) with the only other method of installing Magento, which is the CLI. I can see this leading to some confusion later. 

Basically, in 2.4, the CLI is your ONLY option for installing Magento. The only difference between a contributor and non-contributor is that the former uses a **git-based** install method whereas the latter uses a **Composer-based** install method. 

Technically non-contributors could use an **archive-based** install method, but Magento recommends using Composer as the official method. We decided to de-emphasize the archive-based method in previous iterations of refreshing the Install Guide.

I envision something like this:

- Quick start install (non-contributor, composer-based)
- Quick start install (contributor, git-based)
- Advanced install (all other existing command line install topics)

I also think we need to take a closer look at whether the following **contributor** topics belong in the Install Guide at all:

- Update the Magento application (arguably belongs in the Software Update Guide)
- Add or update components (arguably belongs in the Software Update Guide)
- Change to a released version (i think should stay; it's still git-based install)
- Reinstall the Magento software (i think this topic should be removed entirely)

**Review by jeff-matthews — COMMENTED** [review]
Nice work @shrielenee. In general, I think your changes are solid. Most of my feedback applies to the persona-based approach.

File: src/_data/toc/installation-guide.yml (hunk 5 above)
Comment (shrielenee) [reply src/_data/toc/installation-guide.yml]: OK. After break, I am thoroughly confused about where I was BUT I think I have a good start.

The items you mentioned above I have not yet done are:
* Move these topics (we'll discuss in next meeting): Update the Magento application, Add or update components, Reinstall the Magento software---However, I did remove them from the TOC so they do not appear in the install flow.
* Add content from the child pages to the top level, newly-created Quick start install (contributor) topic. I want to review my preview build with the team to ensure I'm on the right track before smoothing everything out.

**Preview build #137**

Comment (jeff-matthews) [reply src/_data/toc/installation-guide.yml]: @shrielenee, looks good! My suggestions for topic titles aren't very good, so hopefully the team will have better suggestions when we review next week, but you definitely captured the gist of my feedback. Thanks!

File: src/guides/v2.3/install-gde/prereq/install-rabbitmq.md [hunk 6]
```diff
@@ -51,7 +51,7 @@ Refer to the [RabbitMQ/Erlang version matrix](https://www.rabbitmq.com/which-erl
 
 The RabbitMQ server is included on CentOS, but the version is often old. RabbitMQ recommends installing the package from their website.
 
-Refer to the RabbitMQ install page to get the latest supported version. Magento 2.3 and 2.4 support RabbitMQ 3.8.x.
+Refer to the RabbitMQ install page to get the latest supported version. Magento 2.3 and 2.4 supports RabbitMQ 3.8.x.
```
Comment (dobooth) [inline src/guides/v2.3/install-gde/prereq/install-rabbitmq.md:54]: Is 'supports' correct? Seems like singular is correct...

File: src/guides/v2.3/install-gde/trouble/tshoot_access-browser.md [hunk 7]
```diff
@@ -27,9 +27,9 @@ Try the following:
 
 *  Make sure [Apache server rewrites]({{ page.baseurl }}/install-gde/prereq/apache.html) are enabled.
 
-   If Apache server rewrites are set incorrectly, static files are not served from the correct location.
+   If Apache server rewrites are set incorrectly, static files aren't served from the correct location.
```
Comment (dobooth) [inline src/guides/v2.3/install-gde/trouble/tshoot_access-browser.md:30]: I thought we get rid of contractions?

File: src/guides/v2.4/comp-mgr/cli/dev_update-magento.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+menu_title: Update the Magento application
+menu_order: 2
+menu_node:
+functional_areas:
+  - Upgrade
+redirect_from:
+  - guides/v2.4/install-gde/install/cli/dev_update-magento.html
+---
+
+This topic discusses how a contributing developer can update the Magento application without reinstalling it. To perform an upgrade if you're *not* a contributing developer, see [Updating the Magento application and components]({{ page.baseurl }}/comp-mgr/bk-compman-upgrade-guide.html).
```
Comment (dobooth) [inline src/guides/v2.4/comp-mgr/cli/dev_update-magento.md:13]:
```suggestion
This topic discusses how a contributing developer can update the Magento application without reinstalling it. To perform an upgrade if you are *not* a contributing developer, see [Updating the Magento application and components]({{ page.baseurl }}/comp-mgr/bk-compman-upgrade-guide.html).
```

File: src/guides/v2.4/comp-mgr/cli/dev_update-magento.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+To update the Magento software if you're a contributing developer:
```
Comment (dobooth) [inline src/guides/v2.4/comp-mgr/cli/dev_update-magento.md:15]:
```suggestion
To update the Magento software if you are a contributing developer:
```

File: src/guides/v2.4/comp-mgr/dev_add-update.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
+---
+group: software-update-guide
+title: Add or update components
+menu_title: Add or update components
+menu_order: 5
+menu_node:
+functional_areas:
+  - Upgrade
+redirect_from: /guides/v.24/install-gde/install/cli/dev_add-update.html
```
Comment (dobooth) [inline src/guides/v2.4/comp-mgr/dev_add-update.md:9]: This format looks incorrect...at least it is different than the previous one.

File: src/guides/v2.4/comp-mgr/dev_add-update.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+---
+
+A contributing developer updates components by specifying components and their versions in Magento's `composer.json`.
+
+To update components if you're *not* a contributing developer, see [Updating the Magento application and components]({{ page.baseurl }}/comp-mgr/bk-compman-upgrade-guide.html).
```
Comment (dobooth) [inline src/guides/v2.4/comp-mgr/dev_add-update.md:14]:
```suggestion
To update components if you are *not* a contributing developer, see [Updating the Magento application and components]({{ page.baseurl }}/comp-mgr/bk-compman-upgrade-guide.html).
```

File: src/guides/v2.4/comp-mgr/dev_add-update.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+Command usage:
+
+```bash
+composer show --all <vendor>/<name>
+```
+
+For example,
+
+```bash
+composer show --all example/module
```
Comment (dobooth) [inline src/guides/v2.4/comp-mgr/dev_add-update.md:38]: Is this a working example? If not, can it be?

**Review by dobooth — CHANGES_REQUESTED** [review]

File: src/guides/v2.3/install-gde/trouble/tshoot_access-browser.md (hunk 7 above)
Comment (shrielenee) [reply src/guides/v2.3/install-gde/trouble/tshoot_access-browser.md:30]: I do not think this is a hard and fast rule. I also did not copyedit every single page in the install guide. I mostly adding content, removed content, moved content around, and restructured topics and TOC. I will make as many of these changes that I can, however.

File: src/guides/v2.4/comp-mgr/dev_add-update.md (hunk 10 above)
Comment (shrielenee) [reply src/guides/v2.4/comp-mgr/dev_add-update.md:9]: What looks incorrect about it?

File: src/guides/v2.4/comp-mgr/dev_add-update.md (hunk 12 above)
Comment (shrielenee) [reply src/guides/v2.4/comp-mgr/dev_add-update.md:38]: I don't know. Maybe you can try it out!

File: src/guides/v2.4/comp-mgr/dev_add-update.md (hunk 10 above)
Comment (keharper) [reply src/guides/v2.4/comp-mgr/dev_add-update.md:9]: v.24 should be v2.4

**Review by jeff-matthews — APPROVED** [review]

**Pull request merged**